app/database/admin/database_manager/index.ts

```diff
@@ -0,0 +1,365 @@
+                    const databaseConnection = {
+                        actionsEnabled: true,
+                        dbName: displayName,
+                        dbType: DatabaseType.SERVER,
+                        serverUrl: url,
+                    };
+
+                    const dbInstance = await this.createDatabaseConnection({
+                        databaseConnection,
+                        shouldAddToDefaultDB: false,
+                    });
+
+                    return {url, dbInstance};
+                });
+
+                const databaseInstances = await Promise.all(databasePromises);
+                return databaseInstances;
+            }
+            return null;
+        }
+        return null;
+    };
+
+    /**
+     * deleteDBFileOnIOS: Used only on iOS platform to delete a database by its name
+     * @param {string} databaseName
+     */
+    deleteDBFileOnIOS = ({databaseName}: RemoveDatabaseIOS) => {
+        try {
+            deleteIOSDatabase({databaseName});
+        } catch (e) {
+            // console.log('An error occured while trying to delete database with name ', databaseName);
+        }
+    };
+
+    /**
+     * factoryResetOnIOS: Deletes the database directory on iOS
+     * @param {boolean} shouldRemoveDirectory
+     */
+    factoryResetOnIOS = ({shouldRemoveDirectory}: RemoveDatabaseIOS) => {
+        if (shouldRemoveDirectory) {
+            deleteIOSDatabase({shouldRemoveDirectory: true});
+        }
+    };
+
+    /**
+     * removeServerFromDefaultDB : Removes a server record by its url value from the Default database
+     * @param {string} serverUrl
+     * @returns {Promise<void>}
+     */
+    removeServerFromDefaultDB = async ({serverUrl}: { serverUrl: string }) => {
+        try {
+            // Query the servers table to fetch the record with the above displayName
+            const defaultDB = await this.getDefaultDatabase();
+            if (defaultDB) {
+                const serversRecords = await defaultDB.collections.get('servers').query(Q.where('url', serverUrl)).fetch() as IServers[];
+                if (serversRecords.length) {
+                    const targetServer = serversRecords[0];
+
+                    // Perform a delete operation on that record; since there is no sync with backend, we will delete the record permanently
+                    await defaultDB.action(async () => {
+                        await targetServer.destroyPermanently();
+                    });
+
+                    if (Platform.OS === 'ios') {
+                        // Removes the *.db file from the App-Group directory
+                        this.deleteDBFileOnIOS({databaseName: targetServer.displayName});
+                    } else {
+                        // TODO : Perform a similar operation on Android
+                    }
+                }
+            }
+        } catch (e) {
+            // console.error('An error occured while deleting server record ', e);
```

Comment:
Wondering if it's also worth diving into a consistent error-handling / graceful failure for the storage library as well.